pkg/asset/store.go

```diff
@@ -234,12 +274,11 @@ func (s *StoreImpl) Purge(excluded []WritableAsset) error {
 	}
 
 	for _, asset := range toPurge {
-		logrus.Debugf("Purging asset %q", asset.Name())
-		for _, f := range asset.Files() {
-			if err := os.Remove(filepath.Join(s.directory, f.Filename)); err != nil {
-				return errors.Wrapf(err, "failed to remove file %q", f.Filename)
-			}
+		logrus.Debugf("Purging asset %q from disk", asset.Name())
+		if err := DeleteAssetFromDisk(asset, s.directory); err != nil {
+			return err
 		}
 	}
+	s.markedForPurge = []WritableAsset{}
```

Comment:
Are you sure you want this blanket clear?  What about assets that were previously in `s.markedForPurge` but were not purged in this call because they were also in `excluded`?  I'd have expected excluded assets to remain in `s.markedForPurge`.

Reply:
markedforpurge is populated for each Fetch; so purge means clear up. Ideally i wanted to consume each asset when we load, but other people want to keep them around in case of partial failure.